### ISSUE TITLE
bug: Toggle button change background doesn't work if color is set 

### ISSUE THREAD

**Issue by Florian-crg** [ionic-team/ionic-framework#24561]
### Prerequisites

- [x] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

If you   set the property `color` the toggle button css doesn't apply correctly.

For exemple 

`  <ion-toggle slot="end" color="light"></ion-toggle>`

with 

```
 --background: #000 !important;
  --background-checked: #7a49a5 !important;

  --handle-background: #7a49a5 !important;
  --handle-background-checked: #000 !important;
```

will result by   `--background-checked: #7a49a5 !important;` will apply nether on Android nor iOS and `--handle-background-checked: #000 !important;` will not work on android.

### Expected Behavior

The property `color` should be overrode  on both devices.

### Steps to Reproduce

Set the `color` property on `ion-toggle` : `<ion-toggle slot="end" color="light"></ion-toggle>`

Then apply the following css and watch the result on iOS and Andoird, checking and unchecking the toggle button. 

```
ion-toggle {
 --background: #000 !important;
  --background-checked: #7a49a5 !important;
  --handle-background: #7a49a5 !important;
  --handle-background-checked: #000 !important;
}
```

### Code Reproduction URL

https://stackblitz.com/edit/ionic-j5ckyi?file=pages/home/home.ts

### Ionic Info

Ionic:

   Ionic CLI                     : 6.17.0 (/Users/ecovelo/.nvm/versions/node/v12.22.1/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.0.1
   @angular-devkit/build-angular : 13.1.2
   @angular-devkit/schematics    : 12.1.4
   @angular/cli                  : 13.1.2
   @ionic/angular-toolkit        : 4.0.0

Capacitor:

   Capacitor CLI      : 3.2.5
   @capacitor/android : 3.2.5
   @capacitor/core    : 3.2.5
   @capacitor/ios     : 3.2.5

Utility:

   cordova-res (update available: 0.15.4) : 0.15.3
   native-run                             : 1.5.0

System:

   NodeJS : v12.22.1 (/Users/ecovelo/.nvm/versions/node/v12.22.1/bin/node)
   npm    : 8.1.1
   OS     : macOS Monterey

### Additional Information

The stackblitz doesn't reproduce the bug, it is even more glitchy on the stackblitz try.

**Comment by liamdebeasi** [comment]
Thanks for the issue. The `color` property is not intended to be used in conjunction with other CSS Variables that affect the same parts of the component. For example, you set `color="light"` which sets the background color of the toggle, but you also try to override it using the `--background` and `--background-checked` CSS Variables.

When using the `color="light"` property, we set the background values for the toggle according to the theme that is configured in your application (typically these are the CSS Variables in `variables.css` or `global.css`). Since this is configured inside of the toggle component, the styles for `color="light"` received a higher specificity than the styles noted in https://github.com/ionic-team/ionic-framework/issues/24561#issue-1100083528 even when using `!important`.

In other words, you are asking Ionic to set the toggle background, but you are giving it two different values in two different places, so it does not know which one to use. I recommend either using the `color` property and customizing the theme CSS Variables or using the CSS Variables to customize the toggle, but not both.

I am going to close this as this is not a bug in Ionic. Thanks!
